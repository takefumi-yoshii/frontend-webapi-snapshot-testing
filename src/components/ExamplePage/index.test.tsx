import { composeStories } from "@storybook/react";
import * as stories from "./index.stories";
import { waitFor } from "@testing-library/react";
import { mockCreatePost } from "@/services/clients/posts/index.mock";
import { setupMockServer } from "@/tests/jest/msw";
import { renderThenPlay } from "@/tests/jest/storybook";

const server = setupMockServer();
const { Primary } = composeStories(stories);

test("送信値は空白が除去される", async () => {
  // WebAPI の payload を記録するスパイを用意
  const mockFn = jest.fn();
  // MSW のハンドラーを設定
  server.use(mockCreatePost({ mockFn }));
  // Story をレンダリングして PlayFunction を再生する
  await renderThenPlay(Primary);
  // スナップショットテストをアサーションとして使用
  await waitFor(() => expect(mockFn.mock.lastCall[0]).toMatchSnapshot());
});
