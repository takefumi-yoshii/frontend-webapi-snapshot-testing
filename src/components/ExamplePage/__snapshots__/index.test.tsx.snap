// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`送信値は空白が除去される 1`] = `
{
  "payload": {
    "value": "abcd",
  },
}
`;
